core/lib/src/config/secret_key.rs

```diff
@@ -80,6 +98,38 @@ pub struct SecretKey {
     provided: bool,
 }
 
+/// A struct representing encrypted data.
+///
+/// The `Cipher` struct encapsulates encrypted data and provides various
+/// utility methods for encoding and decoding this data in different formats
+/// such as bytes, hexadecimal, and base64.
+///
+/// # Examples
+///
+/// Creating a `Cipher` from bytes:
+/// ```
+/// let data = b"some encrypted data";
+/// let cipher = Cipher::from_bytes(data);
+/// ```
+///
+/// Converting a `Cipher` to a hexadecimal string:
+/// ```
+/// let hex = cipher.to_hex();
+/// ```
+///
+/// Creating a `Cipher` from a base64 string:
+/// ```
+/// let base64_str = "c29tZSBlbmNyeXB0ZWQgZGF0YQ==";
+/// let cipher = Cipher::from_base64(base64_str).unwrap();
+/// ```
+///
+/// Converting a `Cipher` back to bytes:
+/// ```
+/// let bytes = cipher.as_bytes();
+/// ```
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Cipher(Vec<u8>);
```

Comment:
To make Cipher more useful, it could implement `FromForm`, `FromParam`, `FromSegment`, and `FromData`. They should either check the format (which may not be possible, since hex could be re-interpreted as base-64), or always use one specific encoding (and provide a method like `encode` that encodes it using the same encoding).

Reply:
Similarly, we could implement `Serialize` and `Deserialize` if the `serde` feature is enabled.